proc-macros/src/rpc_macro.rs

```diff
@@ -81,11 +82,18 @@ impl RpcMethod {
 			.filter_map(|arg| match arg {
 				syn::FnArg::Receiver(_) => None,
 				syn::FnArg::Typed(arg) => match *arg.pat {
-					syn::Pat::Ident(name) => Some((name, *arg.ty)),
-					_ => panic!("Identifier in signature must be an ident"),
+					syn::Pat::Ident(name) => Some(Ok((name, *arg.ty))),
+					syn::Pat::Wild(_) => Some(Err(syn::Error::new(
+						span,
+						"Method argument names must be valid Rust identifiers; got `_` instead",
+					))),
+					_ => Some(Err(syn::Error::new(
+						span,
+						format!("Unexpected method signature input; got {:?} ", *arg.pat),
+					))),
 				},
 			})
-			.collect();
+			.collect::<Result<Vec<_>, _>>()?;
```

Comment:
```suggestion
			.collect::<Result<_, _>>()?;
```

Or remove `Vec<_>` on line 79.